File: Data Structures & Algorithms/search-2d-matrix/submission-0.ts
class Solution {
    /**
     * @param {number[][]} matrix
     * @param {number} target
     * @return {boolean}
     */

    binarySearch(nums: number[], target: number) : number {
        let l = 0
        let r = nums.length - 1

        while(l <= r) {
            const mid = Math.floor((l+r) / 2)
            if(target > nums[mid]) {
                l = mid + 1
            } else if(target < nums[mid]) {
                r = mid - 1
            } else {
                return mid
            }
        }

        return -1
    }

    searchMatrix(matrix: number[][], target: number): boolean {
        let flag = false
        for(const row of matrix) {
            flag = this.binarySearch(row, target) === -1 ? false : true
            if(flag === true) break;
        }

        return flag
    }
}

File: Data Structures & Algorithms/remove-element/submission-0.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int count = 0;
        int length = nums.length;
        while(count < length) {
            if(nums[count] == val) {
                nums[count] = nums[--length];
            } else {
                count++;
            }
        }

        return length;
    }
}